### PR TITLE
Show docs on Dashboard

### PR DESCRIPTION
**Pull request by MrFlashAccount**
### Pull Request Description

Closes: https://github.com/enso-org/cloud-v2/issues/1445

PR is more or less ready for feedback,CR, and QA.

# Read before looking into:

**Known issues:**
- [x] Need to adjust timings for the animations (they're a bit out of sync).
- [x] After the latest rebase, the side panel might expand while clicking on the sidebar toggle.
- [ ] Images no longer display. 🤦
- [x] Need to restore the functionality around spotlighting a component

Demo:

https://github.com/user-attachments/assets/8cec9ec0-4753-482e-8637-f3857b4396a5

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [ ] The documentation has been updated, if necessary.
- [ ] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [ ] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      [TypeScript](https://github.com/enso-org/enso/blob/develop/docs/style-guide/typescript.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- [ ] Unit tests have been written where possible.

### CODE REVIEW

File: app/gui/src/project-view/components/DocumentationEditor.vue [hunk 1]
```diff
@@ -53,14 +53,17 @@ function useDocumentationImages(
 
   const transformImageUrl = fetcherUrlTransformer(
     async (url: string) => {
+      console.log('url', url)
```
Comment (kazcw) [inline app/gui/src/project-view/components/DocumentationEditor.vue]: Leftover debugging (here and below)

File: app/gui/src/project-view/components/MarkdownEditor/ImagePlugin/LexicalImage.vue [hunk 2]
```diff
@@ -37,6 +37,10 @@ const alt = props.alt ? props.alt : DEFAULT_ALT_TEXT
 onUnmounted(() => {
   if (data.value?.ok) data.value.value.dispose?.()
 })
+
+watch(data, () => {
+  console.log({ data: data.value, title, alt, props })
+})
```
Comment (kazcw) [inline app/gui/src/project-view/components/MarkdownEditor/ImagePlugin/LexicalImage.vue]: Debugging

File: app/gui/src/project-view/util/net/dataServer.ts [hunk 3]
```diff
@@ -203,6 +203,9 @@ export class DataServer extends ObservableV2<DataServerEvents> {
     const segmentsOffset = Path.createSegmentsVector(builder, segmentOffsets)
     const pathOffset = Path.createPath(builder, this.createUUID(path.rootId), segmentsOffset)
     const command = ReadFileCommand.createReadFileCommand(builder, pathOffset)
+
+    console.log('readFile', { path, pathOffset, command })
+
```
Comment (kazcw) [inline app/gui/src/project-view/util/net/dataServer.ts]: Debugging

**Review by kazcw — APPROVED** [review]
LGTM.

FYI, my current WIP will require some changes--that'll be for me to address when my PR is ready, but it'll look like:
- AST: Docs are moving--read documentation from a field of `Ast.Function`.
- So that the rendering is consistent with the project-view's markdown renderer (and uses the same dependencies), reuse the project-view's renderer (with veaury). For example, we are planning to support tables soon; Lexical doesn't support markdown tables, and we found that Lexical's markdown plugin design cannot even be extended to support tables.

**Comment by PabloBuchu** [comment]
1. A though: wouldn't it look better if there wasn't any radius on the right side and the whole row / highlight was touching the bar?
Screenshot 2024-10-29 at 09 00 41

2. Look at the icons (asst panel and docs) when I switch between project view and dashboard
https://github.com/user-attachments/assets/ea7aba8f-db1f-4216-8b19-076fc1988d8c

3. How hard would it be to hide tabs with unavailable content? It could also applies when project has no docs
Screenshot 2024-10-29 at 09 04 52

4. Size and icons between project view bar and dashboard are different.
5. When I click on project which `files` request ends with 404 (that something to fix on backend) I need to close and open asset panel to be able to choose and see docs from another project
Screenshot 2024-10-29 at 09 12 19
6. When I toggle asset panel can we preserve last selected tab? It feels more natural

**Comment by MrFlashAccount** [comment]
> Size and icons between project view bar and dashboard are different.

It's by design. We'll increase the size of the icons on `project view` with updating the icons (out of the scope of this PR)

> How hard would it be to hide tabs with unavailable content? It could also applies when project has no docs

We don't want that because making them always available helps users to learn the interface quicker, and it's closer to how vscode behave


> A though: wouldn't it look better if there wasn't any radius on the right side and the whole row / highlight was touching the bar?

Both look good, but current implementation better matches the design guidelines

**Comment by MrFlashAccount** [comment]
@PabloBuchu your issues are addressed

**Comment by PabloBuchu** [comment]
I think the 5th issue wasn't addressed. Using arrows up and downs I get see their properties without reopning the asset panel. If I do the same with docs and get 404 I need to reopen the panel to clear the state

https://github.com/user-attachments/assets/0b6eb2d0-c8b4-42af-b6f4-e6aed47fdc99

**Comment by MrFlashAccount** [comment]
@PabloBuchu the revalidation is fixed

**Comment by kazcw** [comment]
This doesn't seem to work when using the local backend.

I've pushed an update to the new APIs after #11441. I haven't tested it because my cloud account isn't working today.

**Review by somebody1234 — APPROVED** [review]
CR :white_check_mark:

**Comment by somebody1234** [comment]
partial QA :white_check_mark:

File: app/gui/e2e/dashboard/api.ts [hunk 4]
```diff
@@ -144,6 +145,9 @@ async function mockApiInternal({ page, setupAPI }: MockParams) {
   const addAsset = <T extends backend.AnyAsset>(asset: T) => {
     assets.push(asset)
     assetMap.set(asset.id, asset)
+
+    console.log('addAsset', asset.id)
```
Comment (somebody1234) [inline app/gui/e2e/dashboard/api.ts]: stray `console.log`?

**Pull request merged**